sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -129,6 +129,19 @@ final class DataFrameWriter private[sql](df: DataFrame) {
     this
   }
 
+  @scala.annotation.varargs
+  def bucketBy(numBuckets: Int, colNames: String*): DataFrameWriter = {
```

Comment:
I think this is better as bucketBy(numBuckets, col: String, cols: String*). Need to have at least one column.